replication project.Rmd

```diff
@@ -0,0 +1,243 @@
+
+
+```{r}
+# Create centered variables for Saliance and Variance based on grand mean centering according to author
+
+data <- data  %>% 
+  mutate(salience_mean_c = (salienzmean - 3.84568)) %>% 
+  mutate(var_c = rvar - 21.75423)
+
+# Make male variable numeric
+data <- data %>%
+  mutate(male = as.numeric(male))
+
+# Create a dummy variable for country (transform from character to factor variable)
+data <- data %>%
+  mutate(country = factor(sortcountry, labels = c("AT", "BE","DE-E","DE-W","DK","ES","FI","FR","GR","IT","LU","NL","NO","PT","SE")))
+```
+
+# Run the regression (multilevel logictic regression based on Quasi-penalized likelihood): All contextual variables are standardized mean-centered (as connoted by "z-")
+
+```{r}
+model <- glmmPQL(ext_vote ~ male + age_1 + age_2 + age_4 + edu_1 + edu_2 + farmer_own + worker + retired + unemployed + left_right_scale + eu_neg + z_dem_satis + dispro_elect + federalism + z_asylumseeker + z_unemp + z_asylumseeker:z_unemp + z_replacement + z_replacement:z_unemp + z_replacement:z_asylumseeker + max_er + salience_mean_c + var_c + var_c:salience_mean_c + country - 1, random = ~1|kontext, family = binomial(link = "logit"), data = data, verbose = TRUE)
+# summary(model) # Output too long
+```
+
+
+# Table 1
+![Table 1](figures/table1.png)
+```{r}
+# Create vectors of coefficients, standard errors, and variable names
+
+coefficient <- as.vector(model$coef$fixed) 
+standard_error <- as.vector(sqrt(diag(model$varFix))) 
+
+# Create matrix of coefficients and standard deviations.
+table_matrix <- cbind(coefficient, standard_error)
+
+# Label matrix.
+
+rownames(table_matrix) <- c("Male","18_29 years","30_45 years",">65 years","Education: middle/high","Education: university","Petty Bourgeoisie","Worker","Pensioner","Unemployed","Left-Right","Dissatisfied: EU","Dissatisfied: Democracy","Disproportionality","Decentralization","Asylumseekers", "Unemployment", "Asylumseekers x Unemployment", "Toughness", "Salience", "Variance", "AT", "BE", "DE-E", "DE-W", "DK", "ES", "FI", "FR", "GR", "IT", "NL", "NO", "PT", "SE", "Asylumseekers x Unemployment", "Unemployment benefits x Unemployment", "Unemployment benefits x Asylumseekers", "Variance x Salience")
+
+# Create Table 1 with stargaze
+
+stargazer(table_matrix, title = "TABLE 1 Support for the Extreme Right: Sociodemographics, Attitudes, Country Effects, and Contextual Variables", align = TRUE, table.layout = "t", type = "text", notes = "Logistic multilevel model. PQL2 estimates and model-based standard errors.", style = "ajps", out = "Table 1.html", no.space = TRUE)
+```
+
+# Figure 1 (Conditonal Effects of Unemployment and Immigration)
+![Figure 1](figures/fig1.png)
+```{r}
+#Function with which we will create each graph
+graph <- function(data, ratio, ymin, ymax) {
```

Comment:
is this a clear function name?  Wickham recommends verbs for functions which I think is helpful (http://adv-r.had.co.nz/Style.html).  So it could be something like make_graph()